libs/commons/Logs_.ml

```diff
@@ -211,6 +211,43 @@ let create_formatter opt_file =
   in
   (isatty chan, fmt)
 
+(*****************************************************************************)
+(* Specifying the log level with an environment variable *)
+(*****************************************************************************)
+
+let log_level_of_string_opt str : Logs.level option option =
+  match str with
+  | "app" -> Some (Some App)
+  | "error" -> Some (Some Error)
+  | "warning" -> Some (Some Warning)
+  | "info" -> Some (Some Info)
+  | "debug" -> Some (Some Debug)
+  | "none" -> Some None
+  | _ -> None
+
+(* TODO: document this and handle the interpretation at the time
+   of parsing the command line.
+
+   The PYTEST_ prefix is needed when using pytest because it will unset
+   all other environment variables.
+*)
+let read_level_from_env () =
+  (* left-to-right in order of precedence = from more specific to least
+     specific *)
+  let vars =
+    [
+      "PYTEST_SEMGREP_LOG_LEVEL";
```

Comment:
libs/commons/ is supposed to be independent of semgrep, so would be better to make this an argument of the function
and set those vars in the caller.

Reply:
As per our offline conversation, we'll move this into a module named `Semgrep_logs` or similar.